resources/sdk/purecloudjava/templates/ApiClient.mustache

```diff
@@ -609,6 +611,13 @@ public class ApiClient implements AutoCloseable {
         return future;
     }
 
+    private ApiClientConnectorResponse retryOnFailure(ApiClientConnectorRequest connectorRequest, ApiClientConnectorResponse connectorResponse) {
+        if (connectorResponse.getStatusCode() >= 300) {
+            connectorResponse = retryInvoke(connectorRequest, connectorResponse);
```

Comment:
400 level errors are not retry-able (except 429). I would only retry on a 502, 503, or 504 response codes.

Reply:
Yeah, I would switch on the status code probably

Reply:
@chuckpulfer The current code should only retry if the request is not a 1xx or 2xx **and** there's a retry-after header. Anything that doesn't have a retry-after header won't be retried. I'd prefer for the SDK to be dumb in this case and not check for specific response codes and let the API tell the SDK when it's ok to retry requests by sending the retry-after header. 429 is being checked specifically since we want different behavior in that case. This allows the SDK to be future proof for any changes made to the API regarding when it sends retry-after headers. Do you want the SDK to ignore retry-after headers if the status code isn't 429, 502, 503, or 504?